clicking on subgenus should open taxonomy tab

Mayuresh0072 opened the pull request:
When we click on taxon of rank genus and species, About tab was loading instead of Taxonomy Tab.

kueda reviewed (changes requested):
Good start, @Mayuresh0072! However, you need to fix the issue in #3343 without changing the existing default tab. I think the source of the problem is around https://github.com/inaturalist/inaturalist/blob/main/app/webpack/taxa/show/containers/taxon_page_tabs_container.js#L22, so have a look at that and make sure it handles a subgenus the same way it handles a genus.

Also, in the future please name issue-specific branches starting with the issue number as we've [requested](https://github.com/inaturalist/inaturalist/blob/main/CONTRIBUTING.md#working-on-an-issue). In this case that would be something like `3343-subgenus-taxonomy-tab`. There's no need to change this branch, just something to keep in mind if you decide to work on other issues.

Mayuresh0072 commented:
@kueda I have made the change based on your feedback. Please have a look at it. Thanks.

Mayuresh0072 commented:
Hello @kueda , I have done code changes as per your feedback. Please check it. Thanks.

kueda reviewed (changes requested):
Works, but I'd like a few more changes.

Mayuresh0072 commented:
Hello @kueda , I have removed 'BetweenGenusAndSpecies' as it was redundant. But it should behave as 'aboveGenusTabs' and not as 'genusTabs' because 'genusTabs' have extra tab (similar taxa tab). So, I have changed the condition in 'aboveGenusTabs' which is working perfectly. Second thing, When I ran the command - npx eslint --fix app/webpack/taxa/show/containers/taxon_page_tabs_container.js . It changed the package.json and package-lock.json. I dont know why this is happening. So, I corrected both the files by replacing from the main branch. Sorry for creating more commits , but all changes are correct now in this branch. Please have a look at it. Thanks.

kueda reviewed (approved):
Thanks, @Mayuresh0072, looking good. We'll probably merge and deploy this next week.
